src/content/docs/kubernetes-pixie/kubernetes-integration/get-started/kubernetes-components.mdx

```diff
@@ -0,0 +1,101 @@
+---
+title: New Relic Kubernetes components
+tags:
+  - Integrations
+  - Kubernetes integration
+  - Kubernetes components
+metaDescription: "Learn what components are deployed after installing the Kubernetes integration."
+---
+
+import kubernetesv3Integration from 'images/kubernetes_diagram_v3-integration.png'
+
+The New Relic Kubernetes integration gives you full observability into the health and performance of your cluster, whether you're running Kubernetes on-premises or in the cloud. It gives you visibility into Kubernetes namespaces, deployments, replicasets, nodes, pods, and containers.
+
+## Architecture [#architecture]
+
+The `newrelic-infrastructure` DaemonSet is the main component of the integration. It's divided in three different components: `nrk8s-ksm`, `nrk8s-kubelet`, and `nrk8s-controlplane`. The first is a deployment and the next two are DaemonSets.
+
+Each of the components has two containers:
+
+1. A container for the integration, responsible for collecting metrics.
+2. A container with the New Relic infrastructure agent, which is used to send the metrics to New Relic.
+
+<img
+  title="Diagram showing the integration installed in a 3 nodes cluster"
+  alt="Diagram showing the integration installed in a 3 nodes cluster"
+  src={kubernetesv3Integration}
+/>
+
+### Kube-state-metrics component [#nrk8s-ksm]
+
+We build our cluster state metrics on top of the OSS project[`kube-state-metrics`](https://github.com/kubernetes/kube-state-metrics), which is housed under the Kubernetes organization itself. 
+
+A specific deployment `nrk8s-ksm` takes care of finding KSM and scraping it. With this pod being long-lived and single, it can safely use an endpoints informer to locate the IP of the KSM pod and scrape it. The informer automatically caches the list of informers in the cluster locally and watches for new ones, avoiding storming the API Server with requests to figure out where the pod was located.
+
+See [Bring your own KSM](https://github.com/newrelic/helm-charts/blob/master/charts/nri-bundle/README.md#bring-your-own-ksm) for more information about KSM supported versions.
+
+<Callout variant="important">
+  Use `customAttributes` to add an attribute to amples related to entities that are not strictly tied to a particular node: `K8sNamespaceSample`, `K8sDeploymentSample`, `K8sReplicasetSample`, `K8sDaemonsetSample`, `K8sStatefulsetSample`, `K8sServiceSample`, and `K8sHpaSample`.
+</Callout>
+
+### Kubelet component [#nrk8s-kubelet]
+
+The Kubelet is the “Kubernetes agent”, a service that runs on every Kubernetes node and is responsible for creating the containers as instructed by the control plane. Since it's the Kubelet who partners closely with the Container Runtime, it's the main source of infrastructure metrics for our integration, such as use of CPU, memory, disk, network, etc. Although not thoroughly documented, the Kubelet API is the de-facto standard source for most Kubernetes metrics.
```

Comment:
```suggestion
The Kubelet is the “Kubernetes agent”, a service that runs on every Kubernetes node and is responsible for creating the containers as instructed by the control plane. Since it's the Kubelet who partners closely with the Container Runtime, it's the main source of infrastructure metrics for our integration, such as use of CPU, memory, disk, network, etc. Although not thoroughly documented, the Kubelet API is the de-facto source for most Kubernetes metrics.
```

Reply:
Done.